lib/spack/external/jinja2/sandbox.py

```diff
@@ -23,6 +23,11 @@
 from markupsafe import EscapeFormatter
 from string import Formatter
 
+if sys.version_info >= (3, 3):
```

Comment:
Why check for version instead of use `try-except` here and in other files?  Isn't `collections.abc` introduced in 3.3?

Reply:
Yes, try-except is another valid approach. I personally find this to be cleaner. It boils down to a preference of EAFP vs. LBYL. I'm not sure if PEP-8 has a rule on this? The zen of Python says "explicit is better than implicit", so I tend towards LBYL.

Reply:
Okay. I am under the impression `try-except` is considered the "pythonic" way of doing these kinds of things.  However, I also appreciate being explicit.

That said, doing so makes these changes appear inconsistent with the ones in the other external packages, though I realize they had `try-except` blocks that are now nested in the new `except` clause.

Reply:
Yeah, that's another reason I don't love try-except, because it becomes ugly when you need to do multiple. If you can find something in PEP-8 that dictates which is more Pythonic I'll be happy to change.

Reply:
Ran across a post that says `try-except` is *faster* than using the `if` statement if the `except` *is* the exceptional case.  So it looks like your approach will be the faster of the two for quite some time. ;)

Reply:
The other downside of try-except is that it can catch "exceptions" that you didn't intend (e.g. mistyping a module name).